optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/ProblemBenchmarkResult.java

```diff
@@ -380,7 +380,7 @@ public void registerScale(long registeringEntityCount, long registeringVariableC
             long registeringProblemScale) {
         if (entityCount == null) {
             entityCount = registeringEntityCount;
-        } else if (entityCount.longValue() != registeringEntityCount) {
+        } else if (entityCount != registeringEntityCount) {
```

Comment:
These changes are dangerous. The confuse wether the entityCount is being unboxed or the registeringEntityCount is being boxed (which is a different result when entityCount > 255!).
-1 on these changes. Clarity trumps verbosity for long term projects.

Because this is a big commit, with many changes (and a lot of code just different indented, so any changes in there are invisible to me), I would like to ask to have no other changes to be sneaked in than the lambda change.